### ISSUE TITLE
Custom tokens for placeholders/variables

### ISSUE THREAD

**Issue by fviolette** [expressive-code/expressive-code#207]
Hello,

Is it possible to insert a placeholder within a code block processed by EC?
Following [these guidelines](https://developers.google.com/style/placeholders), I'm thinking of something like:

```yaml
- repo: https://github.com/{{{YOUR_ORGANIZATION}}}/docs
```

where {{{ }}} signifies this has to be processed in a non-literal way, inserting a `<var>` element with possibly a @title picked up from the block options (```yaml var="YOUR_ORGANIZATION:Replace `YOUR_ORGANIZATION` with the name of your organization").

Example:
<img width="990" alt="image" src="https://github.com/expressive-code/expressive-code/assets/19730806/17fb47d0-950c-4fe0-a57b-b84bce21f662">

Thanks.

**Comment by hippotastic** [comment]
Hello François!

This sounds like a great use case for a custom plugin. There are examples on https://expressive-code.com/guides/developing-plugins/#adding-custom-annotations-to-code that show how you can create a plugin that accepts some special syntax inside the code and uses it to add an annotation that can render HTML elements and style them.

It should be possible to adapt this example code to create the functionality you described. Getting parsed information from the meta line is possible using the `codeBlock.metaOptions` API that is available through the context argument passed to each hook.

Hope that helps to point you into the right direction! Let me know if you encounter issues while developing the plugin!

**Comment by fviolette** [comment]
I've made recent progress with this plugin, inspired from https://mkdocs-madlibs.readthedocs.io/en/latest/
Users can update values in code blocks. I'm using localStorage to save placeholder values across codeblocks and pages.

One point: I've noticed the code for a snippet is included in a data-code attribute for the copy to clipboard feature: https://github.com/expressive-code/expressive-code/blob/8e96695c830c7c34071e14b6112789a0aa01849e/packages/%40expressive-code/plugin-frames/src/index.ts#L178

I'm not seeing how to efficiently update blocks so that the copy feature would provide the latest user supplied values.
Updating/refreshing all `@data-code` values based on variables in localStorage sounds overkill and kind of an anti-pattern.

Is this something that the current way EC works can not accomodate easily in your opinion?
